lib/inferno/dsl/fhir_evaluation/rules/all_search_parameters_have_examples.rb

```diff
@@ -58,20 +73,22 @@ def param_is_used?(param, context)
               rescue StandardError => e
                 message = "SearchParameter #{param.url} failed to evaluate due to an error. " \
                           "Expression: #{param.expression}. #{e}"
-                result = EvaluationResult.new(message)
+                result = EvaluationResult.new(message, severity: 'warning', rule: self)
                 context.add_result result
 
-                used = true
+                param_used = true
                 break
               end
 
               if result.present?
-                used = true
+                param_used = true
                 break
               end
             end
-            used
+            param_used
           end
+
+          def add_message; end
```

Comment:
This doesn't feel ideal either but I guess it's fine. This should at least be documented that this is a hack to prevent the `evaluate_fhirpath` from throwing an "undefined method" error since it expects to be called from a Runnable. Otherwise it's not clear why this no-op and apparently unused method is here

Reply:
Agreed. Added comments here.